### PR TITLE
Feature/nil ad email

### PR DESCRIPTION
**Pull request by PaulDoyle-DEFRA**
This change adds a service and Rake task to clear all NCCC contact_email addresses, including a range of known typos.
It also modifies the criteria for sending a letter vs an email to be based on the contact_email being nil, as opposed to matching a specific NCCC email address.
In addition, it removes configuration and exceptions related to the specific assisted digital email address.
https://eaflood.atlassian.net/browse/RUBY-1998

### CODE REVIEW

File: app/views/ad_reminder_letters/bulk.html.erb [hunk 1]
```diff
@@ -1,27 +0,0 @@
-<!doctype html>
```
Comment (Beckyrose200) [inline app/views/ad_reminder_letters/bulk.html.erb:1]: What was this file originally doing? It looks like a template for the reminder letters but was that not done through notify?

File: config/locales/resend_renewal_email.en.yml [hunk 2]
```diff
@@ -2,9 +2,6 @@ en:
   resend_renewal_email:
     messages:
       success: Renewal email sent to %{email}
-      assisted:
-        heading: Sorry, there has been a problem re-sending the renewal email.
-        details: You have requested to resend a renewal email to the contact email address, but this address matches %{email}.
       error:
         heading: We could not send an email to %{email}
```
Comment (Beckyrose200) [inline config/locales/resend_renewal_email.en.yml:6]: Will this catch errors if the email is nil? So will it say "We could not send an email to (nil)"?

File: spec/services/clear_nccc_contact_emails_service_spec.rb [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+      context "lower case (#{base_contact_email})" do
+        let(:contact_email) { base_contact_email.downcase }
+
+        it "sets contact_email to nil" do
+          expect { subject }.to change { registration.reload.contact_email }.to(nil)
+        end
+      end
+
+      # context "mixed case (#{base_contact_email})" do
```
Comment (Beckyrose200) [inline spec/services/clear_nccc_contact_emails_service_spec.rb:84]: Is this meant to be hashed out?

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -26,9 +26,6 @@ jobs:
       AWS_WEEKLY_EXPORT_ACCESS_KEY_ID: 5432
       AWS_WEEKLY_EXPORT_SECRET_ACCESS_KEY: 5432
       AWS_WEEKLY_EXPORT_BUCKET: 5432
-      AWS_LETTERS_EXPORT_ACCESS_KEY_ID: 9012
```
[the remark this answers is not in the document]
Comment (PaulDoyle-DEFRA) [reply .github/workflows/ci.yml:29]: Only one is being removed. It was part of the bulk letter generation process that was used before Notify was adopted, and was overlooked during the cleanup at that time.

File: app/views/ad_reminder_letters/bulk.html.erb (hunk 1 above)
Comment (PaulDoyle-DEFRA) [reply app/views/ad_reminder_letters/bulk.html.erb:1]: Yes, that's done through Notify now. It was previously done within the Rails application, and this was also missed during the cleanup when Notify was adopted.

File: config/locales/resend_renewal_email.en.yml (hunk 2 above)
Comment (Beckyrose200) [reply config/locales/resend_renewal_email.en.yml:6]: I read the rest of the file and can see this isn't the case so ignore me!

Comment (PaulDoyle-DEFRA) [reply config/locales/resend_renewal_email.en.yml:6]: If email is nil, a `MissingContactEmailError` will be raised and `ResendRenewalEmailController` will catch it and display an error message.

File: spec/services/clear_nccc_contact_emails_service_spec.rb (hunk 3 above)
Comment (PaulDoyle-DEFRA) [reply spec/services/clear_nccc_contact_emails_service_spec.rb]: No! Uncommenting....

**Review by Beckyrose200 — APPROVED** [review]

**Pull request merged**